bundles/org.openhab.binding.yamahamusiccast/src/main/java/org/openhab/binding/yamahamusiccast/internal/YamahaMusiccastHandler.java

```diff
@@ -0,0 +1,1331 @@
+    String host;
+    @NonNullByDefault({})
+    public String deviceId = "";
+
+    private YamahaMusiccastStateDescriptionProvider stateDescriptionProvider;
+
+    public YamahaMusiccastHandler(Thing thing, YamahaMusiccastStateDescriptionProvider stateDescriptionProvider) {
+        super(thing);
+        this.stateDescriptionProvider = stateDescriptionProvider;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (!command.equals(RefreshType.REFRESH)) {
+            logger.debug("Handling command {} for channel {}", command, channelUID);
+            channelWithoutGroup = channelUID.getIdWithoutGroup();
+            zone = channelUID.getGroupId();
+            switch (channelWithoutGroup) {
+                case CHANNEL_POWER:
+                    if (command.equals(OnOffType.ON)) {
+                        httpResponse = setPower("on", zone);
+                        tmpString = getResponseCode(httpResponse);
+                        if (!tmpString.equals("0")) {
+                            updateState(channelUID, OnOffType.OFF);
+                        }
+                    } else if (command.equals(OnOffType.OFF)) {
+                        httpResponse = setPower("standby", zone);
+                        tmpString = getResponseCode(httpResponse);
+                        if (!tmpString.equals("0")) {
+                            updateState(channelUID, OnOffType.ON);
+                        }
+                    }
+                    break;
+                case CHANNEL_MUTE:
+                    if (command.equals(OnOffType.ON)) {
+                        httpResponse = setMute("true", zone);
+                        tmpString = getResponseCode(httpResponse);
+                        if (!tmpString.equals("0")) {
+                            updateState(channelUID, OnOffType.OFF);
+                        }
+                    } else if (command.equals(OnOffType.OFF)) {
+                        httpResponse = setMute("false", zone);
+                        tmpString = getResponseCode(httpResponse);
+                        if (!tmpString.equals("0")) {
+                            updateState(channelUID, OnOffType.ON);
+                        }
+                    }
+                    break;
+                case CHANNEL_VOLUME:
+                    tmpString = command.toString();
+                    tmpString = tmpString.replace(".0", "");
+                    try {
+                        volumePercent = Integer.parseInt(tmpString);
+                        volumeAbsValue = (maxVolumeState * volumePercent) / 100;
+                        setVolume(volumeAbsValue, zone, this.host);
+                        if (config.configSyncVolume) {
+                            tmpString = getDistributionInfo(this.host);
+                            @Nullable
+                            DistributionInfo targetObject = new Gson().fromJson(tmpString, DistributionInfo.class);
```

Comment:
See above. Make Gson a field.

Reply:
Prepared changes, will be available with next push